crates/service-client/src/http.rs

```diff
@@ -114,14 +132,34 @@ type Connector = ProxyConnector<HttpsConnector<HttpConnector>>;
 #[derive(Clone, Debug)]
 pub struct HttpClient {
     client: hyper::Client<Connector, Body>,
+    // this can be changed to re-read periodically if necessary
+    request_identity_key: Arc<ArcSwapOption<crate::request_identity::v1::SigningKey>>,
 }
 
 impl HttpClient {
-    pub fn new(client: hyper::Client<Connector, Body>) -> Self {
-        Self { client }
+    pub fn new(
+        client: hyper::Client<Connector, Body>,
+        request_identity_private_key_pem_file: Option<PathBuf>,
+    ) -> Result<Self, HttpClientError> {
+        let request_identity_key = if let Some(request_identity_private_key_pem_file) =
+            request_identity_private_key_pem_file
+        {
+            Arc::new(ArcSwapOption::from(Some(Arc::new(
```

Comment:
Curious, Is`ArcSwapOption::new()` not suitable here? or `from_pointee` perhaps?

Reply:
probably it is yeah :)